packages/cli-repl/src/arg-mapper.ts

```diff
@@ -69,15 +69,23 @@ function mapCliToDriver(options: CliOptions): MongoClientOptions {
       }
     }
   }
-  applyTlsCertificateSelector(options.tlsCertificateSelector, nodeOptions);
+
+  const tlsCertificateSelector = getTlsCertificateSelector(options.tlsCertificateSelector);
+  if (tlsCertificateSelector) {
+    nodeOptions.passphrase = tlsCertificateSelector.passphrase;
+    nodeOptions.pfx = tlsCertificateSelector.pfx;
+  }
+
+  const { version } = require('../package.json');
+  nodeOptions.driverInfo = { name: 'mongosh', version };
+
   return nodeOptions;
```

Comment:
Just a suggestion to simplify this a bit, feel free to take it or leave it (but I think it’s nice that it avoids spelling out `passphrase` and `pfx` explicitly):

```suggestion
  const { version } = require('../package.json');

  return {
    ...nodeOptions,
    ...getTlsCertificateSelector(options.tlsCertificateSelector),
    driverInfo: { name: 'mongosh', version }
  };
```